client/engine/engine.go

```diff
@@ -33,6 +34,11 @@ func (uce *ErrUnhandledChainEvent) Error() string {
 	return fmt.Sprintf("chain event %#v could not be handled by objective %#v due to: %s", uce.event, uce.objective, uce.reason)
 }
 
+type CompletedObjectives struct {
+	l sync.Locker
```

Comment:
Is there a reason we use a `mutex` over a `sync.Map`?

Reply:
Good call. There is no good reason, I think I just ended up here circuitously and forgot about `sync.Map`!

From the godoc of the `sync` package:

> The Map type is optimized for ... when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In [this case], use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.